### ISSUE TITLE
No candidate objects found

### ISSUE THREAD

**Issue by hohenstaufen** [catalyst/moodle-tool_objectfs#318]
Upload task always report "No candidate objects found".

`Execute scheduled task: Object file system upload task (tool_objectfs\task\push_objects_to_storage)
... started 17:23:06. Current memory use 36.6MB.
No candidate objects found.
... used 1 dbqueries
... used 0.18827605247498 seconds
Scheduled task complete: Object file system upload task (tool_objectfs\task\push_objects_to_storage)`

On moodle 3.8.3.
![settings](https://user-images.githubusercontent.com/1156803/81579812-e5f63f80-93ac-11ea-9adb-f2bcdbbbb1c1.png)

![immagine](https://user-images.githubusercontent.com/1156803/81579952-18a03800-93ad-11ea-80b4-c4bac2600581.png)

Any idea on what could be wrong?

**Comment by dennis-ciancoders** [comment]
Any hints on this?, I'm having the exact same problem

**Comment by choma** [comment]
I was having this problem too. I'm using moodle 3.8.2 with DigitalOcean Spaces.
It was correctly uploading all the 9 [tests files](https://github.com/catalyst/moodle-tool_objectfs/tree/master/tests/fixtures), but no other files were uploaded.

I applied the changes on PR #323 by hand, and it started to work. Is it safe to leave it like this? is the travis problem the only reason to not merge this PR?

**Comment by guillefd** [comment]
Yes, this PR #323 is not merged to the stable version (don't know why) but it works ok, I'm using DO spaces on 4 production sites for 6 month now (on moodle 3.9), and works great.
The only detail is that files on spaces are not being deleted when they are deleted locally, have tried the last moodle-tool_objectfs version and still has this issue, but everything else works fine.

**Comment by gcavalcante8808** [comment]
I just created the PR https://github.com/catalyst/moodle-tool_objectfs/pull/413, which is simpler than #323, as it apply the specific 323 DO client fix to the current master.
